src/main/java/bio/terra/cli/command/helperclasses/CommandSetup.java

```diff
@@ -0,0 +1,49 @@
+package bio.terra.cli.command.helperclasses;
+
+import bio.terra.cli.auth.AuthenticationManager;
+import bio.terra.cli.context.GlobalContext;
+import bio.terra.cli.context.WorkspaceContext;
+import java.io.PrintStream;
+import java.util.concurrent.Callable;
+
+public abstract class CommandSetup implements Callable<Integer> {
+  protected GlobalContext globalContext;
+  protected WorkspaceContext workspaceContext;
+
+  // output stream to use for writing command return value
+  protected static final PrintStream OUT = System.out;
+
+  @Override
+  public Integer call() {
+    // read in the global and workspace context
+    globalContext = GlobalContext.readFromFile();
+    workspaceContext = WorkspaceContext.readFromFile();
+
+    // do the login flow if required
+    if (doLogin()) {
+      new AuthenticationManager(globalContext, workspaceContext).loginTerraUser();
+    }
+
+    // execute the command
+    execute();
+
+    // set the command exit code
+    return 0;
```

Comment:
consider having `execute` return the command exit code. Might be good to start using non-zero exit codes for some failure cases..

Reply:
Actually, I think it's better for the command classes to not worry about bash exit codes and just throw exceptions. The exception handling, including updating the error code , happens in `Main` specifically in the `UserActionableAndSystemExceptionHandler` inner class.